### PR TITLE
Raise exception for ddp_cpu not supported for TPUs

### LINKED ISSUE #7810: Doesn't run on CPUs when `accelerator="ddp_cpu"` on TPUs
## 🐛 Bug

Doesn't run on CPUs when `accelerator="ddp_cpu"` is selected for Trainer on TPUs

### To Reproduce

```python
    model = BoringModel()
    trainer = Trainer(
        accelerator="ddp_cpu",
        tpu_cores=8,
        max_epochs=5,
    )
```

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes #7810 

## Before submitting
- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)
- [ ] Did you list all the **breaking changes** introduced by this pull request?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 1]
```diff
@@ -742,6 +742,8 @@ def set_distributed_mode(self, distributed_backend: Optional[str] = None):
 
         # special case with DDP on CPUs
         if self.distributed_backend == "ddp_cpu":
+            if _TPU_AVAILABLE:
+                raise MisconfigurationException("`accelerator='ddp_cpu'` is not supported on TPUs")
```
Comment (awaelchli) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:746]: should we say here on TPU machines / TPU platform?

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/accelerator_connector.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:746]: well on past TPU you can run ddp_pu as The TPU is attached to a "normal" machine, right?

**Comment by Borda** [comment]
@kaushikb11 still draft?

File: pytorch_lightning/trainer/connectors/accelerator_connector.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py]: Updated!

File: tests/accelerators/test_tpu_backend.py [hunk 2]
```diff
@@ -264,3 +264,10 @@ def on_train_end(self, trainer, pl_module):
 
     for param, param_copy in zip(model.parameters(), model_copy.parameters()):
         assert not torch.equal(param.cpu().data, param_copy.data)
+
+
+@RunIf(tpu=True)
+def test_ddp_cpu_not_supported_on_tpus():
+
+    with pytest.raises(MisconfigurationException, match="`accelerator='ddp_cpu'` is not supported on TPUs"):
```
Comment (awaelchli) [inline tests/accelerators/test_tpu_backend.py:272]:
```suggestion
    with pytest.raises(MisconfigurationException, match="`accelerator='ddp_cpu'` is not supported on TPU machines"):
```

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 3]
```diff
@@ -742,6 +742,8 @@ def set_distributed_mode(self, distributed_backend: Optional[str] = None):
 
         # special case with DDP on CPUs
         if self.distributed_backend == "ddp_cpu":
+            if _TPU_AVAILABLE:
+                raise MisconfigurationException("`accelerator='ddp_cpu'` is not supported on TPU machines")
```
Comment (carmocca) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:746]: So if the machine has a TPU at all, ddp cpu does not work? Even if they are disabled/unused?

Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:746]: No idea. Even a few folks at XLA team didn't have an exact solution. You could read more about it in #7810

Comment (tchaton) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:746]:
```suggestion
                raise MisconfigurationException(
                     "`accelerator='ddp_cpu'` is not supported on TPU machines. "
                     "Learn more: https://github.com/PyTorchLightning/pytorch-lightning/issues/7810")
```

**Review by tchaton — APPROVED** [review]
LGTM ! Thanks for checking that

**Review by justusschock — APPROVED** [review]
